### ISSUE TITLE
Discussion: Instead of setting the routes directly to the topology, the generator should return the routes

### ISSUE THREAD

**Issue by arneboockmeyer** [simulate-digital-rail/railway-route-generator#22]
I would like to return the generated routes from the route generator (and the same for all other processing projects) instead of setting them directly to the topology. One potential use case would be to compare projected routes with generated routes.

What do you think, @jeriox @ctiedt @Scretch9 ?

**Comment by ctiedt** [comment]
Could you maybe implement this feature with a new method like this?

```python
def find_routes(self):
    self.generate_routes()
    return self.topology.routes
```

This way our existing workflows should not break and you get an easy way to generate routes directly.

**Comment by arneboockmeyer** [comment]
@ctiedt No, thats not my suggestion. My suggestion is that the generator generates routes and returns them *without* modifying the yaramo-model. It would be possible to add a method like:
```python
def set_routes(self):
    self.topology.routes = self.generate_routes()
```
But *some* should just return the routes.

**Comment by jeriox** [comment]
> @ctiedt No, thats not my suggestion. My suggestion is that the generator generates routes and returns them _without_ modifying the yaramo-model. It would be possible to add a method like:
> 
> ```python
> def set_routes(self):
>     self.topology.routes = self.generate_routes()
> ```
> 
> But _some_ should just return the routes.

I think we didn't really liked the way it was before, because the topology always has an empty list of routes and it was a bit counterintuitive to have the routes laying around separately. Your proposal could ease that, as we could still access the routes via the topology. So I would be okay with that

**Comment by arneboockmeyer** [comment]
Sorry, but I don't understand your point. Before that change I usually had
```python
generator = RouteGenerator()
topology.routes = generator.generate_routes()
```
in my code. Then there is no empty list and no separately storing of the routes - but I could do that, for example to compare routes from the generator and from PlanPro.

I will create a pull request as suggestion.
